From e0297bdb1e3a6a37db8917058b1a933b7109ca73 Mon Sep 17 00:00:00 2001
From: Thomas Bourgoin <thomas.bourgoin@foss.st.com>
Date: Tue, 12 Mar 2024 17:30:43 +0100
Subject: [PATCH 0776/1695] plat-stm32mp1: conf: enable the provisioning driver

Enable the provisioning driver on stm32mp1.

Upstream-Status: Inappropriate [ST downstream https://github.com/STMicroelectronics/optee_os.git 8069935985f23cad590544a49d6fb5c1f678ea29 plat-stm32mp1: conf: enable the provisioning driver]
Signed-off-by: Thomas Bourgoin <thomas.bourgoin@foss.st.com>
Reviewed-on: https://gerrit.st.com/c/mpu/oe/optee/optee_os/+/366594
Change-Id: Iefada19b978b8f1dba368adb668897b02c5eb2c6
Reviewed-on: https://gerrit.st.com/c/mpu/oe/optee/optee_os/+/372974
Reviewed-by: Gatien CHEVALLIER <gatien.chevallier@st.com>
ACI: CITOOLS <MDG-smet-aci-reviews@list.st.com>
Domain-Review: Yann GAUTIER <yann.gautier@foss.st.com>
ACI: CIBUILD <MDG-smet-aci-builds@list.st.com>
---
 core/arch/arm/plat-stm32mp1/conf.mk | 7 +++++++
 1 file changed, 7 insertions(+)

diff --git a/core/arch/arm/plat-stm32mp1/conf.mk b/core/arch/arm/plat-stm32mp1/conf.mk
index e6fc78a2c..967c59e7b 100644
--- a/core/arch/arm/plat-stm32mp1/conf.mk
+++ b/core/arch/arm/plat-stm32mp1/conf.mk
@@ -287,6 +287,7 @@ CFG_STM32_SAES ?= y
 CFG_STM32_TAMP ?= y
 CFG_STM32_TIM ?= y
 CFG_STM32_UART ?= y
+CFG_STM32MP_PROVISIONING ?= y
 CFG_STPMIC1 ?= y
 CFG_SYSCFG ?= y
 CFG_TZC400 ?= y
@@ -308,6 +309,12 @@ $(call force,CFG_STM32_I2C,y)
 $(call force,CFG_STM32_GPIO,y)
 endif
 
+# Provisioning support for BSEC shadow OTP is dedicated to insecure
+# development configuration only.
+ifeq ($(CFG_STM32MP_PROVISIONING),y)
+$(call force,CFG_WARN_INSECURE,y,Required by CFG_STM32MP_PROVISIONING)
+endif
+
 #SAES and CRYP cannot be register at the same time in the crypto framework
 #If CRYP and SAES are enable on STMP32MP13, disable CRYP for safety purpose
 ifeq ($(CFG_STM32MP13),y)
-- 
2.47.2

